File: App/App6/Cliente/src/api/ClienteRoutes.js
import { api_ip } from "../../Config";
const API = api_ip + '/clientes'; // Actualiza la ruta base para los clientes

export const deleteCliente = async (clienteId) => {
  await fetch(`${API}/${clienteId}`, {
    method: "DELETE",
  });
};

export const getClientesByUser = async (userId) => {
  const res = await fetch(`${API}/user/${userId}`, {
    method: "GET",
  });

  return await res.json();
};

export const saveCliente = async (newCliente) => {
  const res = await fetch(`${API}`, {
    method: "POST",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body: JSON.stringify(newCliente),
  });
  return await res.json();
};

export const getCliente = async (clienteId) => {
  const res = await fetch(`${API}/${clienteId}`);
  const cliente = await res.json();
  return cliente;
};

export const updateCliente = async (clienteId, updatedCliente) => {
  const res = await fetch(`${API}/${clienteId}`, {
    method: "PUT",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body: JSON.stringify(updatedCliente),
  });
  return res;
};

File: App/App6/servidor/src/api/cliente_routes.py
from controllers.cliente_controller import Cliente_controller

def clientes_register_routes(app):
    """Define las rutas registradas para los clientes"""
    # Rutas de clientes
    app.route("/clientes/user/<user_id>", methods=['GET'])(Cliente_controller.get_clientes_by_user)
    app.route("/clientes", methods=['POST'])(Cliente_controller.add_cliente)
    app.route("/clientes/<cliente_id>", methods=['GET'])(Cliente_controller.get_cliente)
    app.route("/clientes/<cliente_id>", methods=['PUT'])(Cliente_controller.update_cliente)
    app.route("/clientes/<cliente_id>", methods=['DELETE'])(Cliente_controller.delete_cliente)
